channels/ON-CHAIN.md

```diff
@@ -363,15 +374,144 @@ channel.
+
+The force progress is based on what is considered to be the latest off-chain
+state. We have no way of prooving this is the case so there is a time frame
+provided for the other participant to dispute this. This time frame is
+measured in block height and it works exactly like the `lock_period` does for
+closing sequences - a new force progress can not be posted while a certain
+chain height is not reached. It is worth mentioning is that what is to be
+disputed is the off-chain that the force progress had been based on but not
+the forcing of progress itself. If an older state had been provided by the
+forcing party, the other one can post a newer co-signed off-chain state (via a
+snapshot for example). The co-signed state with the same or greater round will
+replace the on-chain produced one. This is to address the possible attack of
+a malicious actor making a couple of subsequent force pushes so even if one
+started from some old state, eventually a `round` number is produced greater
+than the one co-signed off-chain.
+
+### `channel_force_progress_tx`
+
+The `channel_force_progress_tx` can be included in a block if:
+
+- the last on-chain transaction is NOT a force progress one
+- the last on-chain transaction had been a force progress but the block height
+  timer had passed
+
+It consists of:
+
+- `channel_id`: channel id as recorded on-chain
+- `from`: participant of the channel that posts the force progress transaction
+- `payload`: empty or a transaction proving that the proof of inclusion is part of the channel
+- `solo_payload`: a transaction representing the change to be applied on the
+  channel's state
+- `addresses` - a list of ids for accounts and contracts provided in the proof
+  of inclusion
+- `poi`: proof of inclusion for the old channel's state
+- `ttl`
+- `nonce`: taken from the `from`'s account
+- `fee`
+
+Proof of inclusion represents the internal channel's state. It  
+has to include all accounts, all contracts and their balances.
+Based on this structure - the next `state_hash` is going to be computed
+thus providing insufficient set of accounts and contracts provided will result
+in a different channel `state_hash`.
+
+The payload can be either empty or a signed transaction.
+If the payload is empty - the last on-chain persisted state is used. In this case the proof of inclusion root has of the proof of inclusion must be equal to the one persisted for the channel on-chain. The round being used is the one stored in the channel on-chain.
+If the payload is a transaction it MUST be a channel_offchain_tx. It MUST be co-signed.
+
+An off-chain transaction payload is a valid transaction if it has:
+* `state_hash` equal to the proof of inclusion's root hash. This is a proof
+  that the PoI is a correct one
+* `channel_id` being the same as the transaction `channel_id`
+* `round` being greater to the last on-chain `round` for that channel id.
+
+The solo_payload can not be empty. It MUST be a signed transaction.
```

Comment:
a mutually signed transaction?

Reply:
Nope. The mutually signed would be the `payload` one (representing the state both participants agreed upon).
This `solo_payload` is what would be the next channel state after the force push and it is not expected to be mutually signed (not forbiden, though...)